packages/l/libsdl/xmake.lua

```diff
@@ -23,9 +23,76 @@ package("libsdl")
     elseif is_plat("windows", "mingw") then
         add_syslinks("gdi32", "user32", "winmm", "shell32")
     end
-    add_links("SDL2main", "SDL2")
     add_includedirs("include", "include/SDL2")
 
+    add_configs("use_sdlmain", {description = "Use SDL_main entry", default = true, type = "boolean"})
+
+    on_load(function (package)
+        if package:config("use_sdlmain") then
+            package:add("links", "SDL2main", "SDL2")
+        else
+            package:add("links", "SDL2")
+            package:add("defines", "SDL_MAIN_HANDLED")
+        end
+
+        if package:is_plat("macosx") and package:version():ge("2.0.14") then
+            package:add("frameworks", "CoreHaptics", "GameController")
+        end
+    end)
+
+    on_fetch("linux", "macosx", function (package, opt)
+        if opt.system then
+            -- use sdl2-config
+            local sdl2conf = try {function() return os.iorunv("sdl2-config", {"--version", "--cflags", "--libs"}) end}
+            if sdl2conf then
+                sdl2conf = os.argv(sdl2conf)
+                local sdl2ver = table.remove(sdl2conf, 1)
+                if opt.require_version then
+                    import("core.base.semver")
+                    if not semver.satisfies(sdl2ver, opt.require_version) then
+                        return -- system sdl2 doesn't satisfy version requirements
+                    end
+                end
```

Comment:
please remove this `if end` block, because we need only set result.version, xmake/dev will match version automatically.